Runtime/Scripts/Layers/BinaryMeshLayer/BinaryMeshLayer.cs

```diff
@@ -116,46 +107,52 @@ private IEnumerator DownloadBinaryMesh(TileChange tileChange, System.Action<Tile
 			webRequest.SetRequestHeader("Accept-Encoding", "br");
 #endif
 
-            tiles[tileKey].runningWebRequest = webRequest;
+            tile.runningWebRequest = webRequest;
             yield return webRequest.SendWebRequest();
 
             if (!tiles.ContainsKey(tileKey)) yield break;
 
-            tiles[tileKey].runningWebRequest = null;
+            tile.runningWebRequest = null;
 
             if (webRequest.result != UnityWebRequest.Result.Success)
             {
                 RemoveGameObjectFromTile(tileKey);
-                callback(tileChange);
+                callback?.Invoke(tileChange);
+                yield break;
             }
-            else
-            {
-                byte[] results = webRequest.downloadHandler.data;
 
-                yield return new WaitUntil(() => pauseLoading == false);
-                GameObject newGameobject = CreateNewGameObject(url, results, tileChange);
-                if (newGameobject != null)
-                {
-#if SUBOBJECT
-                    if (hasMetaData)
-                    {
-                        yield return StartCoroutine(LoadMetaData(newGameobject, url));
-                    }
-#endif
+            byte[] results = webRequest.downloadHandler.data;
 
-                    tiles[tileKey].gameObject = newGameobject;
+            yield return new WaitUntil(() => pauseLoading == false);
+            GameObject newGameobject = CreateNewGameObject(url, results, tileChange);
+            
+            if (!newGameobject)
+            {
+                callback?.Invoke(tileChange);
+                yield break;
+            }
 
-                        callback(tileChange);
-                    
-                }
-                else
+            if (tiles.TryGetValue(tileKey, out tile))
+            {
+                if (tile.gameObject) RemoveGameObject(tile.gameObject);
```

Comment:
This is fix #1 where a prior gameobject will be cleaned up, for example when changing the LOD